src/systemtask/SystemTask.cpp

```diff
@@ -190,12 +191,8 @@ void SystemTask::Work() {
       }
     }
 
-    uint32_t systick_counter = nrf_rtc_counter_get(portNRF_RTC_REG);
-    dateTimeController.UpdateTime(systick_counter);
-    batteryController.Update();
-
     monitor.Process();
-
+    dateTimeController.UpdateTime(nrf_rtc_counter_get(portNRF_RTC_REG););
```

Comment:
Semi-colon after `_REG)` shouldn't be there... only the end one ;)

Reply:
Weird, I just checked the branch out and on that line I only see one semicolon, do you know what happened?

Reply:
I confirm, there is an extra ";" after _REG. Forgot to commit/push one last modification ? ;)
Other than that, I've just tested your branch and it works perfectly, the result is very pleasant, Thanks!

Oh and for info, in my code, I explicitely declared the variable 'systick_counter' to tell the reader of the code what's the meaning of `nrf_rtc_counter_get(portNRF_RTC_REG)`. It looks more readable to me.

Reply:
It is more readable - and it would nice if the original two line version stayed.

Reply:
Done! Latest commit keeps the two line version